libs/s25main/Settings.cpp

```diff
@@ -347,8 +347,10 @@ void Settings::LoadIngame()
             const auto* iniWindow = static_cast<const libsiedler2::ArchivItem_Ini*>(settingsIngame.find(window.second));
             if(!iniWindow)
                 continue;
-            windows.persistentSettings[window.first].lastPos.x = iniWindow->getIntValue("pos_x");
-            windows.persistentSettings[window.first].lastPos.y = iniWindow->getIntValue("pos_y");
+            const auto x = windows.persistentSettings[window.first].lastPos.x = iniWindow->getIntValue("pos_x");
+            const auto y = windows.persistentSettings[window.first].lastPos.y = iniWindow->getIntValue("pos_y");
+            windows.persistentSettings[window.first].restorePos.x = iniWindow->getValue("restore_pos_x", x);
+            windows.persistentSettings[window.first].restorePos.y = iniWindow->getValue("restore_pos_y", y);
```

Comment:
It might make sense to reduce the duplication a bit here:
```suggestion
            const auto lastPos = windows.persistentSettings[window.first].lastPos = DrawPoint(iniWindow->getIntValue("pos_x"), iniWindow->getIntValue("pos_y"));
            windows.persistentSettings[window.first].restorePos = DrawPoint(iniWindow->getValue("restore_pos_x", lastPos.x), iniWindow->getValue("restore_pos_y", lastPos.y));
```

The whole idea of the `Point` class is to not have separated x,y-pairs anymore as those may easily lead to mistakes

Reply:
Makes sense.